src/text.h

```diff
@@ -422,33 +422,27 @@ struct Text {
         Backspace(s);
     }
 
-    void ReplaceStr(const wxString &str, const wxString &lstr) {
-        wxString lowert;
-        wxString *fort;
-        const wxString *istr;
-        
-        if (lstr.IsEmpty()) {
-            fort = &t;
-            istr = &str;
+    void ReplaceStr(const wxString &str) {
+        if (sys->casesensitivesearch) {
+            for (int i = 0, j; (j = t.Mid(i).Find(sys->searchstring)) >= 0;) {
+                // does this need WasEdited()?
+                i += j;
+                t.Remove(i, sys->searchstring.Len());
+                t.insert(i, str);
+                i += str.Len();
+            }
         } else {
-            lowert = t.Lower();
-            fort = &lowert;
-            istr = &lstr;
-        }
-
-        for (int i = 0, j; (j = fort->Mid(i).Find(sys->searchstring)) >= 0;) 
-        {
-            // does this need WasEdited()?
-            i += j;
-            fort->Remove(i, sys->searchstring.Len());
-            fort->insert(i, *istr);
-
-            if (!sys->casesensitivesearch) {
+            wxString lowert = t.Lower();

```

Comment:
`Lower()` is an expensive operation, since it allocates a new string and does a unicode sensitive lowercase operation on the whole string.. that should ideally not be done repeatedly.

Reply:
Hello Wouter,
Thanks for your review. This is unavoidable according to my opinion for `lowert` but for `lstr` it is not, so I restored the currently used logic. Is this ok now?
Kind regards
Tobias

Reply:
yeah it is unfortunate wxWidgets doesn't have a case insensitive find, that would at least avoid the allocation.

Reply:
Thanks for your merge!